### PR TITLE
fix: don't leak internal class

### PR DESCRIPTION
**Pull request by ronag**
Don't pass the internal request object, create a "safe" representation. Also don't allow external mutation through `addHeaders`.

### CODE REVIEW

**Comment by ronag** [comment]
The messages here are weird, `:headers` isn't actually headers it's response

**Comment by metcoder95** [comment]
Shall we point to `next`?

File: types/diagnostics-channel.d.ts [hunk 1]
```diff
@@ -9,8 +9,7 @@ declare namespace DiagnosticsChannel {
     completed: boolean;
     method?: Dispatcher.HttpMethod;
     path: string;
-    headers: string;
-    addHeader(key: string, value: string): Request;
+    headers: Array<Buffer>;
```
Comment (ronag) [inline types/diagnostics-channel.d.ts:12]:
```suggestion
    headers: string;
```

File: types/diagnostics-channel.d.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ declare namespace DiagnosticsChannel {
     completed: boolean;
     method?: Dispatcher.HttpMethod;
     path: string;
-    headers: any;
+    headers: string;
```
Comment (ronag) [inline types/diagnostics-channel.d.ts:12]:
```suggestion
    headers: any;
```

**Review by mcollina — CHANGES_REQUESTED** [review]
The goal is to leak the internal class, so it could be used as key in maps etc.

**Comment by ronag** [comment]
That's a bad idea in my opinion and can lead to all kinds of undefined behaviors and breaking. The internal class has no guarantees to its interface.

**Comment by ronag** [comment]
> The goal is to leak the internal class, so it could be used as key in maps etc.

I don't see how this PR would change it. You can still use the passed object as a key.

**Review by mcollina — APPROVED** [review]
lgtm

I missed that

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**